### PR TITLE
Added Gradle build setup for Extension

### PR DESCRIPTION
**Pull request by ahelmbr**
* Added Gradle build and settings files,
* removed unnecessary files (from Maven setup), 
* adapt Readme

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+version = "4.4.4"
+
+
+hivemqExtension {
+    name = "HiveMQ Hello World Extension"
+    author = "HiveMQ"
+    priority = 1000
+    startPriority = 1000
+    mainClass = "$group.HelloWorldMain"
+    sdkVersion = "4.4.3"
```
Comment (SgtSilvio) [inline build.gradle.kts:24]:
```suggestion
    sdkVersion = "$version"
```

File: build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+}
+
+dependencies {
+    hivemqProvided ("ch.qos.logback:logback-classic:1.2.3")
+}
+
```
Comment (SgtSilvio) [inline build.gradle.kts:30]:
```suggestion
```
The logback classes are not needed in the hello world extension

File: build.gradle.kts [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    id("com.hivemq.extension")
+}
+
+repositories {
+    mavenCentral()
+}
+
+
+/* ******************** metadata ******************** */
+
```
Comment (SgtSilvio) [inline build.gradle.kts:12]: these headlines only make sense if the build file is long

File: build.gradle.kts [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+plugins {
+    id("com.hivemq.extension")
+}
+
+repositories {
+    mavenCentral()
+}
+
```
Comment (SgtSilvio) [inline build.gradle.kts:9]:
```suggestion
```
Unnecessary, as the plugin adds maven central automatically.
If you want it for demonstration purposes, please move the `repositories` block down, above the `dependencies` block

File: settings.gradle.kts [hunk 5]
```diff
@@ -0,0 +1,8 @@
+rootProject.name = "HelloWorldExtension"
```
Comment (SgtSilvio) [inline settings.gradle.kts:1]:
```suggestion
rootProject.name = "hivemq-hello-world-extension"
```
Camelcase is not the convention for project names and all our other extensions follow the other pattern

File: settings.gradle.kts [hunk 6]
```diff
@@ -0,0 +1,8 @@
...
+rootProject.buildFileName = "build.gradle.kts"
```
Comment (SgtSilvio) [inline settings.gradle.kts:2]:
```suggestion
```
This is unnecessary, as gradle will look for `build.gradle` and `build.gradle.kts` files by default

File: build.gradle.kts [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+    name = "HiveMQ Hello World Extension"
+    author = "HiveMQ"
+    priority = 1000
+    startPriority = 1000
+    mainClass = "$group.HelloWorldMain"
+    sdkVersion = "$version"
+}
+
+dependencies {
+    testImplementation ("com.hivemq:hivemq-mqtt-client:1.2.1")
```
Comment (SgtSilvio) [inline build.gradle.kts:21]:
```suggestion
    testImplementation("com.hivemq:hivemq-mqtt-client:1.2.1")
```

File: build.gradle.kts [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+    testImplementation("org.junit.jupiter:junit-jupiter-engine:5.6.1")
+    testImplementation("org.junit.jupiter:junit-jupiter-api:5.6.1")
+    testImplementation("com.hivemq:hivemq-testcontainer-junit5:1.1.1")
```
Comment (SgtSilvio) [inline build.gradle.kts:24]:
```suggestion
    testImplementation("org.junit.jupiter:junit-jupiter-api:5.6.1")
    testRuntimeOnly("org.junit.jupiter:junit-jupiter-engine:5.6.1")
    testImplementation("com.hivemq:hivemq-mqtt-client:1.2.1")
    testImplementation("com.hivemq:hivemq-testcontainer-junit5:1.1.1")
```
I prefer grouping dependencies

File: README.adoc [hunk 9]
```diff
@@ -21,17 +21,14 @@ This Hello World extension sets a ClientLifecycleEventListener which logs
...
 to grasp the core concepts of HiveMQ extension development.
 
 == Installation
 
-. Clone this repository into a Java 11 maven project.
-. Run `mvn package` goal from Maven to build the extension.
-. Move the file: "target/hivemq-hello-world-extension-4.4.4-distribution.zip" to the directory: "HIVEMQ_HOME/extensions"
+. Clone this repository into a Java 11 Gradle project.
+. Execute Gradle task 'hivemqExtensionZip' from Gradle to build the extension. (./gradlew hivemqExtensionZip )
+. Move the file: "build/hivemq-extension/HelloWorldExtension-4.4.4.zip" to the directory: "HIVEMQ_HOME/extensions"
```
Comment (SgtSilvio) [inline README.adoc:31]:
```suggestion
. Move the file: "build/hivemq-extension/hivemq-hello-world-extension-4.4.4.zip" to the directory: "HIVEMQ_HOME/extensions"
```

File: assembly.xml [hunk 10]
```diff
@@ -1,51 +0,0 @@
...
-<assembly xmlns="http://maven.apache.org/xsd/assembly-1.1.2.xsd"
-          xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-          xsi:schemaLocation="http://maven.apache.org/xsd/assembly-1.1.2.xsd">
-    <id>distribution</id>
-    <formats>
-        <format>zip</format>
-    </formats>
-    <files>
-        <file>
-            <source>target/LICENSE.txt</source>
```
Comment (SgtSilvio) [inline assembly.xml:27]: The license and readme are now not part of the zip anymore, is this intended?
If not, we should include them as in: https://github.com/hivemq/hivemq-allow-all-extension/blob/master/build.gradle.kts

**Comment by ahelmbr** [comment]
Should we also add the gradle wrapper stuff, like in the allow-all-extension?

**Comment by SgtSilvio** [comment]
> Should we also add the gradle wrapper stuff, like in the allow-all-extension?

Yes, we should add them.
Using the Gradle wrapper gives users the same gradle version as you built it with, so the user can not accidentally use an old/incompatible gradle version.

**Review by SgtSilvio — APPROVED** [review]

**Pull request merged**